app/models/project_services/irker_service.rb

```diff
@@ -0,0 +1,148 @@
+# == Schema Information
+#
+# Table name: services
+#
+#  id         :integer          not null, primary key
+#  type       :string(255)
+#  title      :string(255)
+#  project_id :integer          not null
+#  created_at :datetime
+#  updated_at :datetime
+#  active     :boolean          default(FALSE), not null
+#  properties :text
+#
+
+require 'uri'
+require 'socket'
+
+class IrkerService < Service
+  prop_accessor :colorize_messages, :recipients, :channels
+  validates :recipients, presence: true, if: :activated?
+  validate :check_recipients_count, if: :activated?
+
+  before_validation :get_channels
+
+  def title
+    'Irker (IRC gateway)'
+  end
+
+  def description
+    'Send IRC messages, on update, to a list of recipients through an Irker '\
+    'gateway.'
+  end
+
+  def help
+    msg = 'Recipients have to be specified with a full URI: '\
+    'irc[s]://irc.network.net[:port]/#channel. Special cases: if you want '\
+    'the channel to be a nickname instead, append ",isnick" to the channel '\
+    'name; if the channel is protected by a secret password, append '\
+    '"?key=secretpassword" to the URI.'
+
+    begin
+      default_irc = Gitlab.config.irker['default_irc_uri']
+    rescue Settingslogic::MissingSetting
```

Comment:
This configuration setting (like others) is totally optional. The rescue is here if the 'irker' subsection isn't present at all in the configuration file, but that shouldn't be a problem or stop from using the service, the user would only have to enter full URIs instead of just channel names.